storage/operation/writes_test.go

```diff
@@ -136,6 +141,74 @@ func TestRemove(t *testing.T) {
 	})
 }
 
+func TestRemoveDiskUsage(t *testing.T) {
+	count := 10000
+	wg := sync.WaitGroup{}
+	// 10000 chunk data packs will produce 4 log files
+	// Wait for the 4 log file to be deleted
+	wg.Add(4)
+
+	// Create an event listener to monitor compaction events
+	listener := pebble.EventListener{
+		// Capture when compaction ends
+		WALDeleted: func(info pebble.WALDeleteInfo) {
+			wg.Done()
+		},
+	}
+
+	// Configure Pebble DB with the event listener
+	opts := &pebble.Options{
+		MemTableSize:  64 << 20, // required for rotating WAL
+		EventListener: &listener,
+	}
+
+	dbtest.RunWithPebbleDB(t, opts, func(t *testing.T, r storage.Reader, withWriter dbtest.WithWriter, dir string, db *pebble.DB) {
+		items := make([]*flow.ChunkDataPack, count)
+
+		// prefix is needed for defining the key range for compaction
+		prefix := []byte{1}
+		getKey := func(c *flow.ChunkDataPack) []byte {
+			return append(prefix, c.ChunkID[:]...)
+		}
+
+		for i := 0; i < count; i++ {
+			chunkID := unittest.IdentifierFixture()
+			chunkDataPack := unittest.ChunkDataPackFixture(chunkID)
+			items[i] = chunkDataPack
+		}
+
+		// Insert 100 entities
+		require.NoError(t, withWriter(func(writer storage.Writer) error {
+			for i := 0; i < count; i++ {
+				if err := operation.Upsert(getKey(items[i]), items[i])(writer); err != nil {
+					return err
+				}
+			}
+			return nil
+		}))
+		sizeBefore := getFolderSize(t, dir)
+
+		// Remove all entities
+		require.NoError(t, withWriter(func(writer storage.Writer) error {
+			for i := 0; i < count; i++ {
+				if err := operation.Remove(getKey(items[i]))(writer); err != nil {
+					return err
+				}
+			}
+			return nil
+		}))
+
+		// Trigger compaction
+		require.NoError(t, db.Compact(prefix, []byte{2}, true))
+		wg.Wait()
```

Comment:
Maybe use a timer here so the test doesn't stuck on `wg.Wait()` in case of error.